include/arch/arm/arch.h

```diff
@@ -307,54 +307,58 @@ typedef u32_t k_mem_partition_attr_t;
 #ifdef CONFIG_ARM_USERSPACE
 #ifndef _ASMLANGUAGE
 /* Syscall invocation macros. arm-specific machine constraints used to ensure
- * args land in the proper registers. Currently, they are all stub functions
- * just for enabling CONFIG_USERSPACE on arm w/o errors.
+ * args land in the proper registers.
  */
+extern int _arm_do_syscall(u32_t call_id, u32_t arg1,
+	u32_t arg2, u32_t arg3, u32_t arg4, u32_t arg5, u32_t arg6);
 
 static inline u32_t _arch_syscall_invoke6(u32_t arg1, u32_t arg2, u32_t arg3,
 					  u32_t arg4, u32_t arg5, u32_t arg6,
 					  u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, arg2, arg3, arg4, arg5, arg6, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke5(u32_t arg1, u32_t arg2, u32_t arg3,
 					  u32_t arg4, u32_t arg5, u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, arg2, arg3, arg4, arg5, 0, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke4(u32_t arg1, u32_t arg2, u32_t arg3,
 					  u32_t arg4, u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, arg2, arg3, arg4, 0, 0, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke3(u32_t arg1, u32_t arg2, u32_t arg3,
 					  u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, arg2, arg3, 0, 0, 0, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke2(u32_t arg1, u32_t arg2, u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, arg2, 0, 0, 0, 0, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke1(u32_t arg1, u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(arg1, 0, 0, 0, 0, 0, call_id);
 }
 
 static inline u32_t _arch_syscall_invoke0(u32_t call_id)
 {
-	return 0;
+        return _arm_do_syscall(0, 0, 0, 0, 0, 0, call_id);
 }
 
 static inline int _arch_is_user_context(void)
 {
-	return 0;
+	u32_t control;
```

Comment:
Minor style nit in this function: missing vertical space between variable declaration and after the assembly block, and there's a stray space before the semicolon in the return statement.